crates/provider/src/wallet.rs

```diff
@@ -0,0 +1,111 @@
+use crate::{
+    fillers::{FillProvider, JoinFill, SignerFiller, TxFiller},
+    Provider,
+};
+use alloy_network::{Ethereum, Network, NetworkSigner};
+use alloy_primitives::Address;
+use alloy_transport::Transport;
+
+/// Trait for Providers, Fill stacks, etc, which contain [`NetworkSigner`].
+pub trait WalletProvider<N: Network = Ethereum> {
+    /// The underlying [`NetworkSigner`] type contained in this stack.
+    type Signer: NetworkSigner<N>;
+
+    /// Get a reference to the underlying signer.
+    fn signer(&self) -> &Self::Signer;
+
+    /// Get a mutable reference to the underlying signer.
+    fn signer_mut(&mut self) -> &mut Self::Signer;
+
+    /// Get the default signer address.
+    fn default_signer(&self) -> Address {
+        self.signer().default_signer()
+    }
```

Comment:
hmm, perhaps naming could be confusing here.

`default_signer` vs `signer` perhaps we should include `_address` in the fn name?

Reply:
I want to rename `NetworkSigner` to `NetworkWallet`, and `LocalWallet` to `LocalSigner`, and always use `Wallet` to mean "a collection of credentials" and signer to mean "a single credential". but that is a more extensive refactor

Reply:
opened a new issue #575